WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -0,0 +1,903 @@
+        controller.modalTransitionStyle = .CrossDissolve
+        controller.modalPresentationStyle = .FullScreen
+        presentViewController(controller, animated: true, completion: nil)
+    }
+
+
+    func didTapDiscoverAttribution() {
+        if post?.sourceAttribution == nil {
+            return
+        }
+
+        if let blogID = post?.sourceAttribution.blogID {
+            let controller = ReaderStreamViewController.controllerWithSiteID(blogID, isFeed: false)
+            navigationController?.pushViewController(controller, animated: true)
+            return
+        }
+
+        var path: String?
+        if post?.sourceAttribution.attributionType == SourcePostAttributionTypePost {
+            path = post?.sourceAttribution.permalink
+        } else {
+            path = post?.sourceAttribution.blogURL
+        }
+
+        if let linkURL = NSURL(string: path!) {
+            presentWebViewControllerWithURL(linkURL)
+        }
+    }
+
+
+    func didTapShareButton(sender: UIButton) {
+        ReaderHelpers.sharePost(post!, fromView: sender, inViewController: self)
+    }
+
+
+    func handleBlockSiteNotification(notification:NSNotification) {
+        if let userInfo = notification.userInfo, aPost = userInfo["post"] as? NSObject {
+            if aPost == post! {
+                navigationController?.popViewControllerAnimated(true)
+            }
+        }
+    }
+}
+
+
+// MARK: - WPRichTextView Delegate Methods
+
+extension ReaderDetailViewController : WPRichTextViewDelegate
+{
+
+    public func richTextView(richTextView: WPRichTextView!, didReceiveImageLinkAction imageControl: WPRichTextImage!) {
+        var controller: WPImageViewController
+
+        if WPImageViewController.isUrlSupported(imageControl.linkURL) {
+            controller = WPImageViewController(image: imageControl.imageView.image, andURL: imageControl.linkURL)
+
+        } else if let linkURL = imageControl.linkURL {
+            presentWebViewControllerWithURL(linkURL)
+            return
+
+        } else {
+            controller = WPImageViewController(image: imageControl.imageView.image)
+        }
+
+        controller.modalTransitionStyle = .CrossDissolve
+        controller.modalPresentationStyle = .FullScreen
+
+        presentViewController(controller, animated: true, completion: nil)
+    }
+
+
+    public func richTextView(richTextView: WPRichTextView!, didReceiveLinkAction linkURL: NSURL!) {
+        var url = linkURL
+        if url.host != nil {
+            let postURL = NSURL(string: post!.permaLink)
+            url = NSURL(string: linkURL.absoluteString, relativeToURL: postURL)
+        }
+        presentWebViewControllerWithURL(url)
+    }
+
+}
+
+
+extension ReaderDetailViewController : UIScrollViewDelegate
+{
+    public func scrollViewWillEndDragging(scrollView: UIScrollView, withVelocity velocity: CGPoint, targetContentOffset: UnsafeMutablePointer<CGPoint>) {
+        if UIDevice.isPad() || footerView.hidden || !isLoaded {
```

Comment:
We'll never hide the navbar or the footerview on the iPad, even if the reader is opened in multitasking split view, only on the iPhone.  The rationale is, regardless of model, the iPad always has abundant vertical space.